soc/riscv/riscv-privilege/telink_b9x/soc.c

```diff
@@ -239,3 +281,5 @@ void soc_b9x_restore(void)
 }
 
 SYS_INIT(soc_b9x_init, PRE_KERNEL_1, 0);
+
+SYS_INIT(soc_b9x_check_flash, APPLICATION, 0);
```

Comment:
just check inside soc_b9x_init

Reply:
I can't use it in PRE_KERNEL (we don't have logs at this stage). Moved to earlier stage then APPLICATION - POST_KERNEL